pkg/ansible/watches/watches_test.go

```diff
@@ -601,18 +603,52 @@ func TestMaxWorkers(t *testing.T) {
 			defValue:      1,
 			expectedValue: 3,
 			setEnv:        true,
-			envKey:        "WORKER_MEMCACHESERVICE_CACHE_EXAMPLE_COM",
-			envValue:      3,
+			envVarMap: map[string]int{
+				"WORKER_MEMCACHESERVICE_CACHE_EXAMPLE_COM": 3,
+			},
+		},
+		{
+			name: "max_concurrent_reconciler_%s_%s set to 2, expect 2",
+			gvk: schema.GroupVersionKind{
+				Group:   "cache.example.com",
+				Version: "v1alpha1",
+				Kind:    "MemCacheService",
+			},
+			defValue:      1,
+			expectedValue: 2,
+			setEnv:        true,
+			envVarMap: map[string]int{
+				"MAX_CONCURRENT_RECONCILES_MEMCACHESERVICE_CACHE_EXAMPLE_COM": 2,
+			},
+		},
+		{
+			name: "set multiple env variables",
+			gvk: schema.GroupVersionKind{
+				Group:   "cache.example.com",
+				Version: "v1alpha1",
+				Kind:    "MemCacheService",
+			},
+			defValue:      1,
+			expectedValue: 3,
+			setEnv:        true,
+			envVarMap: map[string]int{
+				"MAX_CONCURRENT_RECONCILES_MEMCACHESERVICE_CACHE_EXAMPLE_COM": 3,
+				"WORKER_MEMCACHESERVICE_CACHE_EXAMPLE_COM":                    1,
+			},
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			os.Unsetenv(tc.envKey)
+			for key, _ := range tc.envVarMap {
+				os.Unsetenv(key)
+			}
 			if tc.setEnv {
-				os.Setenv(tc.envKey, strconv.Itoa(tc.envValue))
+				for key, val := range tc.envVarMap {
+					os.Setenv(key, strconv.Itoa(val))
+				}
```

Comment:
Why you do the loop twice? Would not the following do what you are trying to here?

```go
for key, value := range tc.envVarMap {
	os.Unsetenv(key)
        if if tc.setEnv { 
                 os.Setenv(key, strconv.Itoa(val))
        }
}

```

PS. The CI is failing because of this as well. Check it with `make lint`